phoenix-core/src/main/java/org/apache/phoenix/execute/LiteralResultIterationPlan.java

```diff
@@ -85,6 +85,9 @@ public void close() throws SQLException {
 
             @Override
             public Tuple next() throws SQLException {
+                while (!this.closed && (offset != null && count++ < offset) && tupleIterator.hasNext()) {
+                    tupleIterator.next();
+                }
```

Comment:
This doesn't look right to me. Mind if you correct this and add a correponding test case?